content-fragment-export-to-target/src/aem-cf-console-admin-1/actions/publish/index.js

```diff
@@ -0,0 +1,95 @@
+/*
+* <license header>
+*/
+/*Original logic:*/
+// https://git.corp.adobe.com/CQ/dam-contentfragment/pull/1888/files#diff-6a1ef81fc0da02ed63bfa68d8a57e2ed94dc12b2659d03725d876638892c8b37
+
+const fetch = require('node-fetch')
+const { Core } = require('@adobe/aio-sdk')
+const { errorResponse, getBearerToken, stringParameters, checkMissingRequestInputs, getAemHeaders } = require('../utils');
+const ACTION_ACTIVATE = "Activate";
+
+function getReferencedModels(paths, assets) {
+    const referencedModels = [];
+    const assetsData = assets?.assets || [];
+
+    for (let i = 0; i < assetsData.length; i++) {
+        const asset = assetsData[i];
+        // if server returns a list containing at least one asset which is not yet published
+        // and is not one of the selected nodes to publish
+        if ((!asset?.published || asset?.outdated) && paths.indexOf(asset.path) === -1 && (asset?.type === "contentfragmentmodel" || asset?.type === "asset")) {
+            referencedModels.push(asset?.path);
```

Comment:
It is better to check `asset?.path` at the very beginning, as having it `undefined` doesn't make any sense to proceed.

Reply:
I updated the logic. We can't end up in the situation when we don't have these keys.